gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -264,8 +284,16 @@ public Completable saveFeature(Feature feature) {
 
   /** Clears the currently active project from cache and from local localValueStore. */
   public void clearActiveProject() {
+    localDataStore
+        .getProjectById(localValueStore.getLastActiveProjectId())
+        .flatMapCompletable(localDataStore::removeProject)
+        .subscribe();
     cache.clearActiveProject();
     localValueStore.clearLastActiveProjectId();
     activeProject.onNext(Persistable.notLoaded());
   }
+
+  private boolean isOffline() {
```

Comment:
If we rely on error state we can remove this. Note that Firestore API fails immediately if network connection is unavailable.